pkg/blobstore/cloud/cloud_blob_access.go

```diff
@@ -41,6 +44,14 @@ func (ba *cloudBlobAccess) Get(ctx context.Context, digest digest.Digest) buffer
 		}
 		return buffer.NewBufferFromError(err)
 	}
+
+	go func() {
+		err := ba.touchBlob(ctx, digest)
```

Comment:
This approach has two downsides:
- There is no back-pressure.
- Errors are reported through log.Printf(), meaning they will only be visible in server logs, which are hardly ever checked.

I had exactly the same problem in some of the other code in the tree, which is why I added the `buffer.WithBackgroundTask()` function. With that function you can let a buffer hang right before it's closed and inject a custom error. Could you change this code to use that function instead? I guess you'll need to write something like this:

```go
b, t := buffer.WithBackgroundTask(ba.readBufferFactory.NewBufferFromReader(...))
go func() {
	err := ba.touchBlob(ctx, digest)
	if err != nil {
		err = util.StatusWrap(err, "Failed to refresh blob")
	}
	t.Finish(err)
}()
return b
```

Reply:
Simple enough, done.